annonatate/search.py

```diff
@@ -0,0 +1,106 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+import os
+from bs4 import BeautifulSoup
+
+import pdb
+
+class Search:
+    def __init__(self, request_args, annotations):
+        self.request_args = request_args
+        self.annotations = annotations
+        self.query = self.request_args.get('q')
+        self.allcontent = self.querysearch(self.query)
+        self.tags = self.request_args.get('tag')
+        if self.tags:
+            self.allcontent = self.searchfields(self.allcontent['items'], 'tags', self.tags)
+        self.creator = self.request_args.get('creator')
+        if self.creator:
+            self.allcontent = self.searchfields(self.allcontent['items'], 'creator', self.creator)
+        self.items = self.allcontent['items']
+        self.facets = {}
```

Comment:
Lines 23-28 should probably be a separate function.

Reply:
That makes sense; done